vpr/src/base/SetupGrid.cpp

```diff
@@ -77,17 +77,22 @@ DeviceGrid create_device_grid(std::string layout_name, std::vector<t_grid_def> g
             return build_device_grid(grid_layouts[0], width, height);
         } else {
             //Find the fixed layout close to the target size
-            auto sort_cmp = [](const t_grid_def& lhs, const t_grid_def& rhs) {
-                return lhs.width < rhs.width || lhs.height < rhs.width;
+            std::vector<const t_grid_def *> grid_layouts_view;
```

Comment:
Because the metadata is attached to the t_grid_def, copying the vector would involving copying the metadata as well.  This code avoids the copying of t_grid_def, with a small layer of indirection.  Seems like a reasonable compromise to avoid copying data.